### PR TITLE
Convert some Flow-typed files to TypeScript

### PR DESCRIPTION
**Pull request by jryans**
This converts a portion of Flow-typed files to TypeScript. There are still more to go, but this is already quite a large chunk, so I thought I'd post what I have for now. So far, I've only added basic types where screaming occurred and in nearby areas, so more could be done to type things exhaustively.

Several bugs were also detected by TypeScript (🎉), and fixed:

* `ServerPickerDialog` was testing a field that doesn't exist
* Various props were passed that were no longer / ever used

### CODE REVIEW

File: src/@types/common.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 import { JSXElementConstructor } from "react";
 
 // Based on https://stackoverflow.com/a/53229857/3532235
+// eslint-disable-next-line @typescript-eslint/type-annotation-spacing
```
Comment (t3chguy) [inline src/@types/common.ts:20]: why is this needed now but wasn't before?

File: .eslintrc.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ module.exports = {
             "@typescript-eslint/ban-ts-comment": "off",
 
             "quotes": "off",
+            "indent": "off",
```
Comment (SimonBrandner) [inline .eslintrc.js:33]: I might be missing something :sweat_smile: Why turn off the indent rule? I find it quite helpful when my editor auto-fixes indentation...

Comment (jryans) [reply .eslintrc.js:33]: It's just a reflection of the fact that we have many files that don't follow it due to the way the indent JSX props... I'm open to better suggestions... 😄

File: src/utils/permalinks/Permalinks.ts [hunk 3]
```diff
@@ -74,10 +74,19 @@ const MAX_SERVER_CANDIDATES = 3;
...
 
 export class RoomPermalinkCreator {
+    private _room: Room;
+    private _roomId: string;
+    private _highestPlUserId: string;
+    private _populationMap: { [serverName: string]: number };
+    private _bannedHostsRegexps: RegExp[];
+    private _allowedHostsRegexps: RegExp[];
+    private _serverCandidates: string[];
+    private _started: boolean;
```
Comment (t3chguy) [inline src/utils/permalinks/Permalinks.ts:84]: any reason these remain prefixed? we seem to tend to avoid that in TS where we have an explicit `private` keyword

File: src/utils/permalinks/ElementPermalinkConstructor.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ import PermalinkConstructor, {PermalinkParts} from "./PermalinkConstructor";
  * Generates permalinks that self-reference the running webapp
  */
 export default class ElementPermalinkConstructor extends PermalinkConstructor {
-    _elementUrl: string;
+    private _elementUrl: string;
```
Comment (t3chguy) [inline src/utils/permalinks/ElementPermalinkConstructor.ts:23]: can we un-underscore this please

File: src/utils/StorageManager.ts [hunk 5]
```diff
@@ -36,8 +36,8 @@ function log(msg) {
     console.log(`StorageManager: ${msg}`);
 }
 
-function error(msg) {
-    console.error(`StorageManager: ${msg}`);
+function error(msg, ...args) {
```
Comment (t3chguy) [inline src/utils/StorageManager.ts:39]: types?

File: src/utils/StorageManager.ts [hunk 6]
```diff
@@ -36,8 +36,8 @@ function log(msg) {
...
+    console.error(`StorageManager: ${msg}`, ...args);
 }
 
 function track(action) {
```
Comment (t3chguy) [inline src/utils/StorageManager.ts:43]: type?

File: .eslintrc.js (hunk 2 above)
Comment (jryans) [reply .eslintrc.js:33]: E.g. I could reformat the props to match indent rule, just wasn't sure if we wanted that or not...

File: src/utils/Timer.ts [hunk 7]
```diff
@@ -26,6 +26,13 @@ Once a timer is finished or aborted, it can't be started again
 a new one through `clone()` or `cloneIfRun()`.
 */
 export default class Timer {
+    private _timeout: number;
+    private _timerHandle: NodeJS.Timeout;
+    private _startTs: number;
+    private _promise: Promise<void>;
+    private _resolve: () => void;
+    private _reject: (Error) => void;
```
Comment (t3chguy) [inline src/utils/Timer.ts:34]: same re underscores

File: src/utils/Timer.ts [hunk 8]
```diff
@@ -26,6 +26,13 @@ Once a timer is finished or aborted, it can't be started again
...
+
     constructor(timeout) {
```
Comment (t3chguy) [inline src/utils/Timer.ts:36]: type?

File: src/stores/WidgetEchoStore.ts [hunk 9]
```diff
@@ -65,7 +71,7 @@ class WidgetEchoStore extends EventEmitter {
         return echoedWidgets;
     }
 
-    roomHasPendingWidgetsOfType(roomId, currentRoomWidgets, type: WidgetType) {
+    roomHasPendingWidgetsOfType(roomId, currentRoomWidgets, type?: WidgetType) {
```
Comment (t3chguy) [inline src/stores/WidgetEchoStore.ts:74]: types please

File: src/stores/WidgetEchoStore.ts [hunk 10]
```diff
@@ -16,13 +15,20 @@ limitations under the License.
...
 import EventEmitter from 'events';
+import { IWidget } from 'matrix-widget-api';
 import {WidgetType} from "../widgets/WidgetType";
 
 /**
  * Acts as a place to get & set widget state, storing local echo state and
  * proxying through state from the js-sdk.
  */
 class WidgetEchoStore extends EventEmitter {
+    private _roomWidgetEcho: {
```
Comment (t3chguy) [inline src/stores/WidgetEchoStore.ts:26]: do we need the underscore?

File: src/stores/TypingStore.ts [hunk 11]
```diff
@@ -25,15 +25,23 @@ const TYPING_SERVER_TIMEOUT = 30000;
  * Tracks typing state for users.
  */
 export default class TypingStore {
+    private _typingStates: {
```
Comment (t3chguy) [inline src/stores/TypingStore.ts:28]: do we need the underscore?

File: src/indexing/EventIndexPeg.ts [hunk 12]
```diff
@@ -27,11 +27,16 @@ import {SettingLevel} from "../settings/SettingLevel";
 
 const INDEX_VERSION = 1;
 
-class EventIndexPeg {
+export class EventIndexPeg {
+    public index: EventIndex;
+    public error: Error;
+
+    private _supportIsInstalled: boolean;
```
Comment (t3chguy) [inline src/indexing/EventIndexPeg.ts:34]: do we need the underscore?

File: src/indexing/EventIndexPeg.ts [hunk 13]
```diff
@@ -27,11 +27,16 @@ import {SettingLevel} from "../settings/SettingLevel";
...
+
     constructor() {
         this.index = null;
-        this._supportIsInstalled = false;
         this.error = null;
+        this._supportIsInstalled = false;
     }
```
Comment (t3chguy) [inline src/indexing/EventIndexPeg.ts:40]: these could all be initialised above

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.tsx [hunk 14]
```diff
@@ -23,10 +23,24 @@ import Field from "../../../elements/Field";
...
+    autoLaunchSupported: boolean;
+    warnBeforeExit: boolean;
+    warnBeforeExitSupported: boolean;
+    alwaysShowMenuBarSupported: boolean;
+    alwaysShowMenuBar: boolean;
+    minimizeToTraySupported: boolean;
+    minimizeToTray: boolean;
+    autocompleteDelay: string,
+    readMarkerInViewThresholdMs: string,
+    readMarkerOutOfViewThresholdMs: string,
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/PreferencesUserSettingsTab.tsx:39]: we should probably agree on `;` vs `,` - we seem to have more `;`s, you seem to mix them

File: src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx [hunk 15]
```diff
@@ -26,16 +25,28 @@ import StyledRadioGroup from '../../../elements/StyledRadioGroup';
...
+import { replaceableComponent } from "../../../../../utils/replaceableComponent";
 
-@replaceableComponent("views.settings.tabs.room.SecurityRoomSettingsTab")
-export default class SecurityRoomSettingsTab extends React.Component {
-    static propTypes = {
-        roomId: PropTypes.string.isRequired,
-    };
+type JoinRule = "public" | "knock" | "invite" | "private";
+type GuestAccess = "can_join" | "forbidden";
+type History = "invited" | "joined" | "shared" | "world_readable";
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx:32]: HistoryVisibility maybe?

File: .eslintrc.js (hunk 2 above)
Comment (jryans) [reply .eslintrc.js:33]: (The rule is already off for JS files above, I'm just copying it because the problem files are now in TSX.)

Comment (SimonBrandner) [reply .eslintrc.js:33]: Changing the indentation would be great imo, but I am not sure what others think :man_shrugging:

File: src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx (hunk 15 above)
Comment (t3chguy) [inline src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx:32]: Enums might be nicer

File: src/ScalarAuthClient.ts [hunk 16]
```diff
@@ -31,6 +30,12 @@ const imApiVersion = "1.1";
...
 export default class ScalarAuthClient {
+    private apiUrl: string;
+    private uiUrl: string;
+    private scalarToken: string;
+    private termsInteractionCallback: TermsInteractionCallback;
+    private isDefaultManager: boolean;
+
     constructor(apiUrl, uiUrl) {
         this.apiUrl = apiUrl;
         this.uiUrl = uiUrl;
```
Comment (t3chguy) [inline src/ScalarAuthClient.ts:41]: this could be `constructor(private apiUrl: string, private uiUrl: string) {` and then lines 33, 34 could be removed

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx [hunk 17]
```diff
@@ -63,14 +63,14 @@ function parseIntWithDefault(val, def) {
...
 
-export class BannedUser extends React.Component {
-    static propTypes = {
-        canUnban: PropTypes.bool,
-        member: PropTypes.object.isRequired, // js-sdk RoomMember
-        by: PropTypes.string.isRequired,
-        reason: PropTypes.string,
-    };
+interface IBannedUserProps {
+    canUnban: boolean;
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:67]: the PropType says this is optional, is this change intentional?

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx [hunk 18]
```diff
@@ -63,14 +63,14 @@ function parseIntWithDefault(val, def) {
...
+    member: RoomMember;
+    by: string;
+    reason: string;
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:70]: the PropType says this is optional, is this change intentional?

File: src/components/views/settings/EventIndexPanel.tsx [hunk 19]
```diff
@@ -104,6 +111,8 @@ export default class EventIndexPanel extends React.Component {
 
     _onManage = async () => {
```
Comment (t3chguy) [inline src/components/views/settings/EventIndexPanel.tsx:112]: can we un-underscore and `private` these?

File: src/components/views/rooms/EventTile.tsx [hunk 20]
```diff
@@ -169,101 +173,130 @@ const MAX_READ_AVATARS = 5;
...
-        enableFlair: PropTypes.bool,
-
-        // whether or not to show read receipts
-        showReadReceipts: PropTypes.bool,
-    };
+export default class EventTile extends React.Component<IProps, IState> {
+    private _suppressReadReceiptAnimation: boolean;
+    private _isListeningForReceipts: boolean;
+    private _tile = React.createRef();
+    private _replyThread = React.createRef();
```
Comment (t3chguy) [inline src/components/views/rooms/EventTile.tsx:299]: underscores + `private` = sad :(

File: src/components/views/rooms/EventTile.tsx [hunk 21]
```diff
@@ -169,101 +173,130 @@ const MAX_READ_AVATARS = 5;
...
+    // that we can tell when it changes.
+    eventSendStatus?: string;
+
+    // the shape of the tile. by default, the layout is intended for the
+    // normal room timeline.  alternative values are: "file_list", "file_grid"
+    // and "notif".  This could be done by CSS, but it'd be horribly inefficient.
+    // It could also be done by subclassing EventTile, but that'd be quite
+    // boiilerplatey.  So just make the necessary render decisions conditional
+    // for now.
+    tileShape?: string;
```
Comment (t3chguy) [inline src/components/views/rooms/EventTile.tsx:248]: enum/string const type please

File: src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.tsx [hunk 22]
```diff
@@ -26,14 +25,23 @@ import {formatBytes, formatCountLong} from "../../../../utils/FormattingUtils";
 import EventIndexPeg from "../../../../indexing/EventIndexPeg";
 import {SettingLevel} from "../../../../settings/SettingLevel";
 
+interface IProps {
+    onFinished: (boolean) => {},
```
Comment (t3chguy) [inline src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.tsx:29]: according to EventIndexPanel this method is void and does not care for a boolean

File: src/Terms.ts [hunk 23]
```diff
@@ -27,6 +27,10 @@ export class TermsNotSignedError extends Error {}
  * require agreement from the user before the user can use that service.
  */
 export class Service {
+    public serviceType: string;
+    public baseUrl: string;
+    public accessToken: string;
```
Comment (t3chguy) [inline src/Terms.ts:32]: these could be inlined into c'tor as

`constructor(public serviceType: string, public baseUrl: string, public accessToken: string) {}`

File: src/components/views/rooms/MessageComposer.tsx [hunk 24]
```diff
@@ -98,39 +103,39 @@ const EmojiButton = ({addEmoji}) => {
...
-    static propTypes = {
-        roomId: PropTypes.string.isRequired,
-    }
+interface IUploadButtonProps {
+    roomId: string;
+}
+
+class UploadButton extends React.Component<IUploadButtonProps> {
+    private _uploadInput = React.createRef<HTMLInputElement>();
+    private _dispatcherRef: string;
```
Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.tsx:112]: private members don't need underscores

Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.tsx:112]: this could probably be `private readonly`

File: src/components/views/rooms/MessageComposer.tsx [hunk 25]
```diff
@@ -98,39 +103,39 @@ const EmojiButton = ({addEmoji}) => {
...
+    private onUploadClick = () => {
         if (MatrixClientPeg.get().isGuest()) {
             dis.dispatch({action: 'require_registration'});
             return;
         }
         this._uploadInput.current.click();
     }
 
-    onUploadFileInputChange(ev) {
+    private onUploadFileInputChange = (ev) => {
```
Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.tsx:138]: type please

File: src/components/views/rooms/MessageComposer.tsx [hunk 26]
```diff
@@ -98,39 +103,39 @@ const EmojiButton = ({addEmoji}) => {
...
-        this._uploadInput = createRef();
         this._dispatcherRef = dis.register(this.onAction);
     }
 
     componentWillUnmount() {
         dis.unregister(this._dispatcherRef);
     }
 
-    onAction = payload => {
+    private onAction = payload => {
```
Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.tsx:124]: I think this can be of type `ActionPayload`

File: src/components/views/rooms/MessageComposer.tsx [hunk 27]
```diff
@@ -57,18 +66,14 @@ function SendButton(props) {
...
-
 const EmojiButton = ({addEmoji}) => {
     const [menuDisplayed, button, openMenu, closeMenu] = useContextMenu();
 
     let contextMenu;
     if (menuDisplayed) {
         const buttonRect = button.current.getBoundingClientRect();
         const EmojiPicker = sdk.getComponent('emojipicker.EmojiPicker');
-        contextMenu = <ContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu} catchTab={false}>
+        contextMenu = <ContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu}>
```
Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.tsx:76]: this got renamed to `managed` and should still be `managed={false}` - my bad at some point probably

**Review by t3chguy — CHANGES_REQUESTED** [review]
MatrixGlob.ts, Mjolnir.ts, ListRule.ts, BanList.ts have some more private underscored members

This is great, can't wait for us to be rid of prop-types and `.js` files :D

File: .eslintrc.js (hunk 2 above)
Comment (jryans) [reply .eslintrc.js:33]: Okay, I agree it's better fixed, I'll work on it. 🙂

Comment (jryans) [reply .eslintrc.js]: Okay, fixed now and rule restored.

File: src/utils/permalinks/Permalinks.ts (hunk 3 above)
Comment (jryans) [reply src/utils/permalinks/Permalinks.ts]: Have fixed all except `_serverCandidates` since there's a getter with same name.

File: src/indexing/EventIndexPeg.ts (hunk 12 above)
Comment (jryans) [reply src/indexing/EventIndexPeg.ts:34]: Here we do, as there's a getter with the same name.

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.tsx (hunk 14 above)
Comment (jryans) [reply src/components/views/settings/tabs/user/PreferencesUserSettingsTab.tsx:39]: Ah just a mistake, will fix to `;`. Probably we should lint it...

File: src/ScalarAuthClient.ts (hunk 16 above)
Comment (jryans) [reply src/ScalarAuthClient.ts:41]: Ah cool, hadn't seen that syntax yet, will change.

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx (hunk 18 above)
Comment (jryans) [reply src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:70]: No, will fix.

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx (hunk 17 above)
Comment (jryans) [reply src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:67]: No, will fix.

File: src/@types/common.ts (hunk 1 above)
Comment (jryans) [reply src/@types/common.ts:20]: Ah, an artifact of trying to do too many things... It's needed on a different branch which changes lint settings. I'll remove it here.

File: src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.tsx (hunk 22 above)
Comment (jryans) [reply src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.tsx]: Well, the `boolean` is the usual one for modals, which is where this function ends up going. It's true the `EventIndexPanel` caller doesn't currently use that value. Fixed the return syntax.

**Comment by jryans** [comment]
Okay, I think I've now addressed all notes from the first round. 😄

File: src/ScalarAuthClient.ts [hunk 28]
```diff
@@ -31,7 +30,11 @@ const imApiVersion = "1.1";
...
 
 export default class ScalarAuthClient {
-    constructor(apiUrl, uiUrl) {
+    private scalarToken: string;
+    private termsInteractionCallback: TermsInteractionCallback;
+    private isDefaultManager: boolean;
+
+    constructor(private apiUrl: string, private uiUrl: string) {
         this.apiUrl = apiUrl;
         this.uiUrl = uiUrl;
```
Comment (t3chguy) [inline src/ScalarAuthClient.ts:39]: with this syntax you do not need the `this.X = X;`

File: src/ScalarAuthClient.ts [hunk 29]
```diff
@@ -100,7 +103,7 @@ export default class ScalarAuthClient {
         }
     }
 
-    _getAccountName(token) {
+    private getAccountName(token) {
```
Comment (t3chguy) [inline src/ScalarAuthClient.ts:106]: type please

File: src/ScalarAuthClient.ts [hunk 30]
```diff
@@ -125,8 +128,8 @@ export default class ScalarAuthClient {
         });
     }
 
-    _checkToken(token) {
-        return this._getAccountName(token).then(userId => {
+    private checkToken(token) {
```
Comment (t3chguy) [inline src/ScalarAuthClient.ts:131]: type please

File: src/ScalarAuthClient.ts [hunk 31]
```diff
@@ -243,7 +246,7 @@ export default class ScalarAuthClient {
     disableWidgetAssets(widgetType: WidgetType, widgetId) {
```
Comment (t3chguy) [inline src/ScalarAuthClient.ts:246]: type please

File: src/Terms.ts [hunk 32]
```diff
@@ -32,13 +32,33 @@ export class Service {
      * @param {string} baseUrl The Base URL of the service (ie. before '/_matrix')
      * @param {string} accessToken The user's access token for the service
      */
-    constructor(serviceType, baseUrl, accessToken) {
+    constructor(public serviceType: string, public baseUrl: string, public accessToken: string) {
         this.serviceType = serviceType;
         this.baseUrl = baseUrl;
         this.accessToken = accessToken;
     }
```
Comment (t3chguy) [inline src/Terms.ts:39]:
```suggestion
    constructor(public serviceType: string, public baseUrl: string, public accessToken: string) {
    }
```

File: src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.tsx [hunk 33]
```diff
@@ -26,14 +25,23 @@ import {formatBytes, formatCountLong} from "../../../../utils/FormattingUtils";
 import EventIndexPeg from "../../../../indexing/EventIndexPeg";
 import {SettingLevel} from "../../../../settings/SettingLevel";
 
+interface IProps {
+    onFinished: (boolean) => void;
```
Comment (t3chguy) [inline src/async-components/views/dialogs/eventindex/ManageEventIndexDialog.tsx:29]: this defined a parameter of name `boolean` but with no type
image

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx [hunk 34]
```diff
@@ -107,37 +109,37 @@ export class BannedUser extends React.Component {
...
+    private populateDefaultPlEvents(eventsSection, stateLevel, eventsLevel) {
         for (const desiredEvent of Object.keys(plEventsToShow)) {
             if (!(desiredEvent in eventsSection)) {
                 eventsSection[desiredEvent] = (plEventsToShow[desiredEvent].isState ? stateLevel : eventsLevel);
             }
         }
     }
 
-    _onPowerLevelsChanged = (value, powerLevelKey) => {
+    private onPowerLevelsChanged = (value, powerLevelKey) => {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:142]: types please

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx [hunk 35]
```diff
@@ -182,7 +184,7 @@ export default class RolesRoomSettingsTab extends React.Component {
         });
     };
 
-    _onUserPowerLevelChanged = (value, powerLevelKey) => {
+    private onUserPowerLevelChanged = (value, powerLevelKey) => {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:187]: types please

File: src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx [hunk 36]
```diff
@@ -26,64 +25,92 @@ import StyledRadioGroup from '../../../elements/StyledRadioGroup';
...
         );
         const encrypted = MatrixClientPeg.get().isRoomEncrypted(this.props.roomId);
         this.setState({joinRule, guestAccess, history, encrypted});
-        const hasAliases = await this._hasAliases();
+        const hasAliases = await this.hasAliases();
         this.setState({hasAliases});
     }
 
-    _pullContentPropertyFromEvent(event, key, defaultValue) {
+    private pullContentPropertyFromEvent(event, key, defaultValue) {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx:104]: types please

File: src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx [hunk 37]
```diff
@@ -149,7 +175,7 @@ export default class SecurityRoomSettingsTab extends React.Component {
         });
     };
 
-    _onRoomAccessRadioToggle = (roomAccess) => {
+    private onRoomAccessRadioToggle = (roomAccess) => {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx:178]: type please

File: src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx [hunk 38]
```diff
@@ -195,7 +221,7 @@ export default class SecurityRoomSettingsTab extends React.Component {
         });
     };
 
-    _onHistoryRadioToggle = (history) => {
+    private onHistoryRadioToggle = (history) => {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx:224]: type please

File: src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx [hunk 39]
```diff
@@ -206,11 +232,11 @@ export default class SecurityRoomSettingsTab extends React.Component {
         });
     };
 
-    _updateBlacklistDevicesFlag = (checked) => {
+    private updateBlacklistDevicesFlag = (checked) => {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/SecurityRoomSettingsTab.tsx:235]: type please

File: src/components/views/settings/tabs/user/PreferencesUserSettingsTab.tsx [hunk 40]
```diff
@@ -128,38 +142,38 @@ export default class PreferencesUserSettingsTab extends React.Component {
...
     };
 
-    _onReadMarkerOutOfViewThresholdMs = (e) => {
+    private onReadMarkerOutOfViewThresholdMs = (e) => {
         this.setState({readMarkerOutOfViewThresholdMs: e.target.value});
         SettingsStore.setValue("readMarkerOutOfViewThresholdMs", null, SettingLevel.DEVICE, e.target.value);
     };
 
-    _renderGroup(settingIds) {
+    private renderGroup(settingIds) {
```
Comment (t3chguy) [inline src/components/views/settings/tabs/user/PreferencesUserSettingsTab.tsx:176]: types please

File: src/utils/Timer.ts [hunk 41]
```diff
@@ -26,44 +26,49 @@ Once a timer is finished or aborted, it can't be started again
...
-        this._onTimeout = this._onTimeout.bind(this);
-        this._setNotStarted();
+    private timerHandle: NodeJS.Timeout;
+    private startTs: number;
+    private promise: Promise<void>;
+    private resolve: () => void;
+    private reject: (Error) => void;
+
+    constructor(private timeout: number) {
+        this.timeout = timeout;
```
Comment (t3chguy) [inline src/utils/Timer.ts:36]:
```suggestion
    constructor(private timeout: number) {
```

File: src/utils/Timer.ts [hunk 42]
```diff
@@ -26,44 +26,49 @@ Once a timer is finished or aborted, it can't be started again
...
+            this.setNotStarted();
         } else {
-            const delta = this._timeout - elapsed;
-            this._timerHandle = setTimeout(this._onTimeout, delta);
+            const delta = this.timeout - elapsed;
+            this.timerHandle = setTimeout(this.onTimeout, delta);
         }
     }
 
     changeTimeout(timeout) {
```
Comment (t3chguy) [inline src/utils/Timer.ts:63]: type please

File: src/utils/permalinks/Permalinks.ts [hunk 43]
```diff
@@ -96,130 +105,130 @@ export class RoomPermalinkCreator {
...
-                this._updateAllowedServers();
-                this._updateHighestPlUser();
-                this._updatePopulationMap();
-                this._updateServerCandidates();
+                this.updateAllowedServers();
+                this.updateHighestPlUser();
+                this.updatePopulationMap();
+                this.updateServerCandidates();
                 return;
             case "m.room.power_levels":
```
Comment (t3chguy) [inline src/utils/permalinks/Permalinks.ts:170]: any chance you fancy switching these to EventType enum from js-sdk :D?

File: src/utils/permalinks/Permalinks.ts [hunk 44]
```diff
@@ -96,130 +105,130 @@ export class RoomPermalinkCreator {
...
-                this._updateHighestPlUser();
-                this._updateServerCandidates();
+                this.updateHighestPlUser();
+                this.updateServerCandidates();
                 return;
         }
     }
 
-    onMembership(evt, member, oldMembership) {
+    private onMembership(evt, member, oldMembership) {
```
Comment (t3chguy) [inline src/utils/permalinks/Permalinks.ts:177]: types please

**Review by t3chguy — COMMENTED** [review]
moar

**Review by t3chguy — APPROVED** [review]
🚢 It
